services/harmony/app/models/job.ts

```diff
@@ -500,16 +501,39 @@ export class Job extends DBRecord implements JobRecord {
     let query;
 
     if (includeLabels) {
-      query = tx(Job.table)
-        .select(`${Job.table}.*`, tx.raw(`STRING_AGG(${LABELS_TABLE}.value, ',' order by value) AS label_values`))
-        .leftOuterJoin(`${JOBS_LABELS_TABLE}`, `${Job.table}.jobID`, '=', `${JOBS_LABELS_TABLE}.job_id`)
-        .leftOuterJoin(`${LABELS_TABLE}`, `${JOBS_LABELS_TABLE}.label_id`, '=', `${LABELS_TABLE}.id`)
-        .where(setTableNameForWhereClauses(Job.table, constraints.where))
-        .groupBy(`${Job.table}.id`)
-        .orderBy(
-          constraints?.orderBy?.field ?? 'createdAt',
-          constraints?.orderBy?.value ?? 'desc')
-        .modify((queryBuilder) => modifyQuery(queryBuilder, constraints));
+      if (constraints.labels) { // Requesting to limit the jobs based on the provided labels
```

Comment:
Is it possible that someone might want to filter on labels, but doesn't need the labels included in the result? I think these are separate concerns.

Reply:
There is currently not a use case for it or code that would request to not include labels but does filter on labels. If we do need that functionality at some point later we should add the code in then to handle that case too.

Reply:
(Would need to add `includeLabels` as a parameter.)